Sources/Plasma/PubUtilLib/plAvatar/plAvatarClothing.cpp

```diff
@@ -71,26 +71,21 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
 
 #include "pfMessage/plClothingMsg.h"
 
-plClothingItem::plClothingItem() : fGroup(0), fTileset(0), fType(0), fSortOrder(0),
-                                   fThumbnail(nil), fAccessory(nil)
-{   
-    int i;
-    fTextures.Reset();
-    fElementNames.Reset();
-    fElements.Reset();
-    
-    for (i = 0; i < 3; i++)
-    {
-        fDefaultTint1[i] = fDefaultTint2[i] = 255;
-    }
-    for (i = 0; i < kMaxNumLODLevels; i++)
-        fMeshes[i] = nil;
+plClothingItem::plClothingItem()
+    : fGroup(), fType(), fTileset(), fSortOrder(), fThumbnail(),
+      fAccessory()
+{
+    std::fill(std::begin(fMeshes), std::end(fMeshes), nullptr);
+    std::fill(std::begin(fDefaultTint1), std::end(fDefaultTint1), 255);
+    std::fill(std::begin(fDefaultTint2), std::end(fDefaultTint2), 255);
```

Comment:
I wonder if it would make more sense to use an initializer list for this (optionally, converting to `std::array`). Without looking in the IDE or the header, I thought these were vectors. My thought was, "wait, why isn't this a no-op?" - but they are C arrays. 😥

Reply:
An initializer list would require the source file know the length of the arrays though (true even for `std::array`).  This way, the array size is not assumed, and the compiler can still optimize it to a `memset`.